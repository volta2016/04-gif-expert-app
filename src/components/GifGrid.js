import React from 'react'

const GifGrid = ({category}) => {
    
    const getGifs = async() => {
        const url = 'https://api.giphy.com/v1/gifs/search?q=Rick+and+Morty&limit=10&api_key=rEsg35kMJgcpu89M3sbOsXXeTd5BLxc6';
        const resp = await fetch(url);
        const {data} = await resp.json();

        console.log(data);	


		}
		getGifs();

    return (
        <div>
            <h3>{ category }</h3>
        </div>
    )
}

export default GifGrid
