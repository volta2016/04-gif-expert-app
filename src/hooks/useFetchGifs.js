import { useState } from "react"


export const useFetchGifs = () => {
    
    const [state, setState] = useState({
        data: [],
        loading: true
    });

    setTimeout(() => {
        setState({
            data: [1,3,4,5],
            loading: false
        })
    }, 3000);

    

    return state;// {data[], loading: true};

}