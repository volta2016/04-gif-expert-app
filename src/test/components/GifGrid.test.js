import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import GifGrid from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");
//mock falsify something

describe("debe de hacer un snapshot de <GifGrid/>", () => {
	const category = "One Punch";

	test("debe de mostrar el componente correctamente", () => {
		useFetchGifs.mockReturnValue({
			data: [],
			loading: true,
		});

		const wrapper = shallow(<GifGrid category={category} />);
		expect(wrapper).toMatchSnapshot();
	});

	test("debe de mostrar items cuando se cargan imágenes con useFetchGifs", () => {
		const gifs = [
			{
				id: "ABC",
				url: "https://localhost/random/object.jpg",
				title: "Title pass test",
			},
			{
				id: "ABC",
				url: "https://localhost/random/object.jpg",
				title: "Title pass test",
			},
		];

		useFetchGifs.mockReturnValue({
			data: gifs,
			loading: false,
		});

		const wrapper = shallow(<GifGrid category={category} />);
		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find("p").exists()).toBe(false); //el parrot no existe que esperaría yo toBe eso no debería de existir.
		expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
	});
});
