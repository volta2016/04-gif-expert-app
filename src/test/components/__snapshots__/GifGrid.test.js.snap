// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`debe de hacer un snapshot de <GifGrid/> debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="card-grid animate__animated animate__fadeIn"
  >
    One Punch
     
    <span
      aria-label="stars"
      role="img"
    >
      ðŸŒŸ
    </span>
  </h3>
  <div
    className="spinner"
  />
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`debe de hacer un snapshot de <GifGrid/> debe de mostrar items cuando se cargan imÃ¡genes con useFetchGifs 1`] = `
<Fragment>
  <h3
    className="card-grid animate__animated animate__fadeIn"
  >
    One Punch
     
    <span
      aria-label="stars"
      role="img"
    >
      ðŸŒŸ
    </span>
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Title pass test"
      url="https://localhost/random/object.jpg"
    />
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Title pass test"
      url="https://localhost/random/object.jpg"
    />
  </div>
</Fragment>
`;
